api/src/pcapi/routes/backoffice/user_deletion/blueprint.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+from flask import flash
+from flask import redirect
+from flask import render_template
+from flask import url_for
+from markupsafe import Markup
+from sqlalchemy.exc import IntegrityError
+from werkzeug.exceptions import NotFound
+
+from pcapi import settings
+from pcapi.core.users import models as users_models
+from pcapi.models import db
+from pcapi.repository import atomic
+from pcapi.repository import mark_transaction_as_invalid
+from pcapi.routes.backoffice import blueprint
+from pcapi.routes.backoffice import utils
+
+from . import forms
+
+
+logger = logging.getLogger(__name__)
+
+
+@blueprint.backoffice_web.route("/delete", methods=["GET"])
+@utils.custom_login_required(redirect_to=".home")
+def get_user_deletion_form() -> str:
+    form = forms.UserDeletionForm()
+    return render_template("admin/users_deletion.html", form=form)
+
+
+@blueprint.backoffice_web.route("/delete", methods=["POST"])
+@atomic()
+@utils.custom_login_required(redirect_to=".home")
+def delete_user() -> utils.BackofficeResponse:
+    if not settings.ENABLE_USER_DELETION:
+        raise NotFound()
+
+    form = forms.UserDeletionForm()
+    if not form.validate():
+        flash(utils.build_form_error_msg(form), "warning")
+        return redirect(url_for("backoffice_web.get_user_deletion_form"), code=303)
+
+    email = form.email.data
+    user = users_models.User.query.filter_by(email=email).one_or_none()
+    if not user:
+        flash(
+            Markup("L'adresse mail <b>{email}</b> n'a pas été trouvée.").format(email=email),
+            "warning",
+        )
+        return redirect(url_for("backoffice_web.delete_user"), code=303)
+
+    try:
+        db.session.delete(user)
+        db.session.flush()
+    except IntegrityError as e:
+        logger.error(e)
```

Comment:
[détail] Si on souhaite logger à ce niveau de log-là, autant logger le traceback complet avec `exception() : 

```suggestion
        logger.exception(e)
```

... mais souhaite-t-on vraiment voir cette erreur dans Sentry ? Est-ce qu'on en fera quelque chose ? Je pense que non. Je suggère donc de logger seulement au niveau `info`. Ça permettra éventuellement de débugger si quelqu'un nous dit "je n'arrive pas à supprimer l'utilisateur X, savez-vous pourquoi ?", mais ça ne nous polluera pas Sentry.